### PR TITLE
Moved frobenius_polynomial and optimized it

### PR DESCRIPTION
**Pull request by kevinywlui**
The ap is now computed more efficiently.

### CODE REVIEW

File: sage_modabvar/abvar.py [hunk 1]
```diff
@@ -445,6 +447,36 @@ def label(self):
...
         degen = str(self.degen_t()).replace(' ','')
         return '%s%s'%(self.newform_label(), degen)
+    
+    def newform(self, names=None):
+        """
+        Return the newform `f` such that this abelian variety is isogenous to
+        the newform abelian variety `A_f`. If this abelian variety is not
+        simple, raise a ValueError.
+
+        INPUT: "names" - if the newform has coefficients in a number field,
```
Comment (williamstein) [inline sage_modabvar/abvar.py]: The input and output blocks are formatted incorrectly.  Look it up in the Sage dev guide -- there is a template there.

File: sage_modabvar/abvar.py [hunk 2]
```diff
@@ -445,6 +447,36 @@ def label(self):
...
+        then a generator name must be specified.
+
+        OUTPUT: newform
+
+        EXAMPLES::
```
Comment (williamstein) [inline sage_modabvar/abvar.py:464]: blank line after ::, I think.

File: sage_modabvar/abvar.py [hunk 3]
```diff
@@ -445,6 +447,36 @@ def label(self):
...
+            sage: from sage_modabvar import J0
+            sage: J0(11).newform()
+            q - 2*q^2 - q^3 + 2*q^4 + q^5 + O(q^6)
+            sage: f = J0(23).newform(names='a')
+            sage: AbelianVariety(f) == J0(23)
+            True
+
+        The following fails since `J_0(33)` is not simple::
```
Comment (williamstein) [inline sage_modabvar/abvar.py:474]: blank line

File: sage_modabvar/abvar.py [hunk 4]
```diff
@@ -2022,6 +2054,44 @@ def _ambient_hecke_matrix_on_modular_symbols(self, n):
         self.__ambient_hecke_matrix_on_modular_symbols[n] = T
         return T
 
+    def frobenius_polynomial(self, p):
+        """
+        Computes the frobenius polynomial at `p`. If this abelian variety is
+        not simple, a ValueError will be raised.
```
Comment (williamstein) [inline sage_modabvar/abvar.py]: "If this abelian variety is not simple, a ValueError will be raised." -- don't use the passive voice.   How about "... not simple, raises a ValueError."

File: sage_modabvar/abvar.py [hunk 5]
```diff
@@ -2022,6 +2054,44 @@ def _ambient_hecke_matrix_on_modular_symbols(self, n):
...
+
+        INPUT:
+
+        - ``p`` -- prime number
+
+        OUTPUT:
+            a monic integral polynomial
+
+        EXAMPLES::
+            from sage_modabvar import AbelianVariety
```
Comment (williamstein) [inline sage_modabvar/abvar.py]: Missing `sage:` prompt at the beginning of the line.  Did you run the tests?  Did they work with the missing prompt?

File: sage_modabvar/abvar.py [hunk 6]
```diff
@@ -2022,6 +2054,44 @@ def _ambient_hecke_matrix_on_modular_symbols(self, n):
...
+            sage: f = Newform('39b','a')
+            sage: A=AbelianVariety(f)
+            sage: A.frobenius_polynomial(5)
+            x^4 + 2*x^2 + 25
+
+            from sage_modabvar import J0
```
Comment (williamstein) [inline sage_modabvar/abvar.py]: missing prompt

**Comment by kevinywlui** [comment]
I think I fixed the issues here. Not entirely sure about the output block though. The blank line after the double colon issue seem to happen elsewhere as well, so at some point I'll fix those.

**Pull request merged**